content/docs/command-reference/repro.md

```diff
@@ -110,7 +110,8 @@ up-to-date and only execute the final stage.
   reproduced independently.
 
 - `-R`, `--recursive` - determines the stages to reproduce by searching each
-  target directory (if any) and their subdirectories.
+  target directory and its subdirectories. This option only has an effect if a
+  directory is given as one of the `targets`.
```

Comment:
This change doesn't clarify that this option "expands what the targets argument accepts (directory paths)." If you're unsure what the issue requirements mean please ping me @imhardikj, thanks.

Reply:
The same applies for -P, it should clarify the same idea.

Reply:
> expands what the targets argument accepts (directory paths)

This refers to the top description of `targets`: "Stage or path to dvc.yaml or .dvc file to reproduce" <-- notice it doesn't include directories, nor are dirs mentioned as targets anywhere in the desc. That's why clarification is needed.

Reply:
Yeah I realize my mistake. `-R` option will expand paths if they are given as targets. Then it will be search for `dvc.yaml` or `.dvc` in that path (and folders inside it), and then reproduce them.
In case of `-P`, I checked that it works without any targets given. So basically it expands the path of the DVC project and find all targets to reproduce from it.

Reply:
No, you're not getting me yet. Maybe I'm not being clear? I wasn't talking about "expanding paths". Once more, these options "expand what the `targets` argument accepts" i.e. usually targets only accept "Stage or path to dvc.yaml or .dvc file to reproduce", right? It doesn't say that paths to directories are accepted. With these options, paths to directories are also accepted.

Like I mentioned, please ping me if you're not 99% sure you're understanding me. Please go over this again and lmk when to review. Thanks

Reply:
> It turns out that with these options, paths to directories are also accepted even if it's not stated in the Synopsis/usage.

So another possible solution is to change the usage text to include "directories" but then we have to clarify somewhere that dirs are only used by the command if -PR are used) so I'm not sure which way is better/ easier to explain. Let's stick to the current approach unless you have a strong opinion.

Reply:
Continued in https://github.com/iterative/dvc.org/pull/1861#pullrequestreview-517073211. Resolving here